src/block_iterator.rs

```diff
@@ -91,42 +94,27 @@ impl<B: BlockLike> BlockIterator<B> {
             block,
             off_off: idx,
             first_key,
+            row_attributes,
         }
     }
 
     fn advance(&mut self) {
         self.off_off += 1;
     }
 
-    fn load_at_current_off(&self) -> Option<KeyValueDeletable> {
+    fn load_at_current_off(&self) -> Result<Option<KeyValueDeletable>, SlateDBError> {
         if self.off_off >= self.block.offsets().len() {
-            return None;
+            return Ok(None);
         }
         let off = self.block.offsets()[self.off_off];
         let off_usz = off as usize;
         // TODO: bounds checks to avoid panics? (paulgb)
         let mut cursor = self.block.data().slice(off_usz..);
-        let overlap_len = cursor.get_u16() as usize;
-        let rest_len = cursor.get_u16() as usize;
-        let rest_key = cursor.slice(..rest_len);
-        cursor.advance(rest_len);
-        let value_len = cursor.get_u32();
-
-        let v = if value_len == TOMBSTONE {
-            ValueDeletable::Tombstone
-        } else {
-            let value = cursor.slice(..value_len as usize);
-            ValueDeletable::Value(value)
-        };
 
-        let mut key = BytesMut::with_capacity(overlap_len + rest_len);
-        key.extend_from_slice(&self.first_key[..overlap_len]);
-        key.extend_from_slice(&rest_key);
-
-        Some(KeyValueDeletable {
-            key: key.into(),
-            value: v,
-        })
+        match decode_row_v0(&self.first_key, &self.row_attributes, &mut cursor) {
```

Comment:
you can just use `?` here

Reply:
I can't because I need to wrap the row in `Some(row)`